test/elastic_record/index/documents_test.rb

```diff
@@ -99,12 +99,36 @@ def test_expired_scroll_error
     )
 
     scroll_enumerator.request_more_hits
-    index.connection.json_delete '/_search/scroll', scroll_id: scroll_enumerator.scroll_id
+    index.delete_scroll(scroll_enumerator.scroll_id)
     assert_raises ElasticRecord::ExpiredScrollError do
       scroll_enumerator.request_more_hits
     end
   end
 
+  def test_each_slice
+    10.times { |i| index.index_document("bob#{i}", color: 'red') }
+    batches = []
+
+    scroll_enumerator = index.build_scroll_enumerator(search: {'query' => {query_string: {query: 'color:red'}}}, batch_size: 1)
+
+    scroll_enumerator.each_slice do |slice|
+      batches << slice
+    end
+
+    assert_equal 10, batches.size
+  end
+
+  def test_delete_scroll
+    scroll_enumerator = index.build_scroll_enumerator(search: {'query' => {query_string: {query: ''}}})
+    scroll_enumerator.initial_search_response
+
+    index.delete_scroll(scroll_enumerator.scroll_id)
+    delete = index.connection.deferred_actions.first
+
+    assert_equal :json_delete, delete.method
+    assert_equal ['/_search/scroll', {scroll_id: scroll_enumerator.scroll_id}], delete.args
+  end
```

Comment:
Too much internal testing here...

Reply:
The deferred action thing makes it difficult to assert anything about the request made without doing this or mocks. Would you just prefer not to test?

Reply:
I'll just disable using the wrapper to disable deferring and use webmock.

Reply:
uh please don't use webmock

Reply:
Ok... so like; just delete it? I don't think I can do anything else to assert any behavior.

Reply:
Guess, I can either make another request and catch the error or have to trust the stats request.